### ISSUE TITLE
Planning: discretized_path_ and frenet_path_ are inserted twice into path_data_history_ if FLAGS_use_front_axe_center_in_path_planning is true

### ISSUE THREAD

**Issue by jilinzhou** [ApolloAuto/apollo#10906]
**Describe the bug**
Here is code snippet when the optimized path are set in path_data:

```
      auto frenet_frame_path =
          ToPiecewiseJerkPath(opt_l, opt_dl, opt_ddl, path_boundary.delta_s(),
                              path_boundary.start_s());

      // TODO(all): double-check this;
      // final_path_data might carry info from upper stream
      PathData path_data = *final_path_data;
      path_data.SetReferenceLine(&reference_line);
      path_data.SetFrenetPath(std::move(frenet_frame_path));
      if (FLAGS_use_front_axe_center_in_path_planning) {
        auto discretized_path = DiscretizedPath(
            ConvertPathPointRefFromFrontAxeToRearAxe(path_data));
        path_data = *final_path_data;
        path_data.SetReferenceLine(&reference_line);
        path_data.SetDiscretizedPath(discretized_path);
      }
```
The call to path_data.SetFrenetPath(std::move(frenet_frame_path)) inserts the discretized_path into path_data_history as the following:

```
bool PathData::SetFrenetPath(FrenetFramePath frenet_path) {
 ...
  path_data_history_.emplace_back(discretized_path_, frenet_path_);
  return true;
}
```

If FLAGS_use_front_axe_center_in_path_planning ==true, the call to path_data.SetDiscretizedPath(discretized_path) will again insert the converted discretized_path into path_data_history as the following:

```
bool PathData::SetDiscretizedPath(DiscretizedPath path) {
 ...
  path_data_history_.emplace_back(discretized_path_, frenet_path_);
  return true;
}
```

**Comment by ycool** [comment]
Nice catch.  Thanks a lot for this great help :)

**Comment by jilinzhou** [comment]
No problem. Thanks for all your hard work!

**Comment by jilinzhou** [comment]
I tried the fix today on the vehicle and found it does not work as expected. Then I noticed Function ConvertPathPointRefFromFromAxeToRearAxe() relies on path_data.discretized_path_ which does not get set until PathData::SetFrenetPath(FrenetFramePath frenet_path) is called first.

```
      if (FLAGS_use_front_axe_center_in_path_planning) {
        auto discretized_path = DiscretizedPath(
            ConvertPathPointRefFromFrontAxeToRearAxe(path_data));
        path_data = *final_path_data;
        path_data.SetReferenceLine(&reference_line);
        path_data.SetDiscretizedPath(discretized_path);
      }
```
I also realized the original code is fine because of  the new assignment of `path_data = *final_path_data` inside this condition check in which the default copy assignment operator is called. Therefore there is no 2nd insertion into path_data_history_ for the same path_data.
